tests/test_optimizers.py

```diff
@@ -12,14 +12,14 @@
 model(input).sum().backward()
 
 
-def make_test(optim_cls, exp_frame_cnt=1, **kwargs):
+def make_test(optim_cls, exp_frame_cnt=1, nopython=True, **kwargs):
     opt = optim_cls(model.parameters(), **kwargs)
 
     def test_fn(self):
         nonlocal opt
         counter = torchdynamo.testing.CompileCounter()
 
-        with torchdynamo.optimize(counter):
+        with torchdynamo.optimize(counter, nopython=nopython):
```

Comment:
If you remove nopython=True, Adam will continue to work. I just tried this locally.